#1-4 Task. Punko Pavel

PashaPunko opened the pull request:
https://travis-ci.org/github/PashaPunko/js-assignments/builds/706069585

Commit:
Update the links

Commit:
Tasks#1-4.PashaPunko

PashaPunko committed:
Update .travis.yml

AlexMokin commented on task/04-arrays-tasks.js:
```diff
@@ -519,7 +541,8 @@ function group(array, keySelector, valueSelector) {
 function selectMany(arr, childrenSelector) {
-    throw new Error('Not implemented');
+    arr = arr.map(x=>childrenSelector(x));
+    return arr.reduce((acc, val) => acc.concat(val), []);
```
this is alright, but you could use childrenSelector inside reduce and do the same in one pass

AlexMokin commented on task/04-arrays-tasks.js:
```diff
@@ -536,7 +559,8 @@ function selectMany(arr, childrenSelector) {
 function getElementByIndexes(arr, indexes) {
-    throw new Error('Not implemented');
+    if (indexes.length == 1) return arr[indexes[0]];
+    else return getElementByIndexes(arr[indexes[0]], indexes.slice(1))
```
works but it's better to not use recursion if possible. Try to do it with reduce function

AlexMokin commented on task/04-arrays-tasks.js:
```diff
@@ -559,7 +583,10 @@ function getElementByIndexes(arr, indexes) {
  * 
  */
 function swapHeadAndTail(arr) {
-    throw new Error('Not implemented');
```
looks like a "swap two numbers without using a temp" problem. Only you're not saving anything here, each map creates new array.  Try to find simpler solution. Hint: use concat and slice

AlexMokin commented:
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

PashaPunko commented:
https://travis-ci.org/github/PashaPunko/js-assignments/builds/706069585

AlexMokin commented:
Please carefully review the commetns and make the fixes according to the comments.

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

PashaPunko commented:
Good evening. There was a wrong link on travis, probably. I fixed it and each tasks are complete. So if you mean this issue in your comments, can I mark this PR as ready for review?

AlexMokin commented:
I mean i left notes in the comments. The code must be fixed according with the comments. You can find comments on the **conversation** and **Files changed** tabs

E.g.

![image](https://user-images.githubusercontent.com/7272208/87226980-3fc0b900-c3a0-11ea-9f57-6c55f3c066d0.png)

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

PashaPunko commented:
> I mean i left notes in the comments. The code must be fixed according with the comments. You can find comments on the conversation and Files changed tabs
> E.g.
> 
> I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

![screencapture-github-AisBrestEDU-js-assignments-pull-333-2020-07-11-15_42_06](https://user-images.githubusercontent.com/43681652/87228214-5965fe80-c3a8-11ea-97d9-b75313e6d7f7.png)

That's a problem. As you may see at the screen short, there is no comments you left in conversation.

PashaPunko committed:
Update 01-strings-tasks.js

I updated this file according to comment I saw at the screenshot in conversation

PashaPunko committed:
Update 02-numbers-tasks.js

I updated this file according to comment I saw at the screenshot in conversation

PashaPunko committed:
Update 04-arrays-tasks.js

I updated this file according to comment I saw at the screenshot in conversation

PashaPunko committed:
Update 02-numbers-tasks.js

PashaPunko committed:
Update 02-numbers-tasks.js

PashaPunko committed:
Update 02-numbers-tasks.js

AlexMokin commented on task/01-strings-tasks.js:
```diff
@@ -201,7 +201,19 @@ function extractEmails(str) {
+    const V_LINE = '\u2502';
+    let str = '';
+    if (width > 1 && height>1){
+        str = LT_ANGLE + new String(H_LINE).repeat(width-2) + RT_ANGLE+'\n';
```
H_LINE is already a string you don't need to wrap it in new String()

AlexMokin reviewed (commented):
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

Commit:
task 1-4 fixing

AlexMokin commented:
accepted
